### PR TITLE
[v9] UIP-767: new spacing classes

### PR DESCRIPTION
**Pull request by reywright**
## Description
- Added new directions (`x` and `y`) so that you don't have to specify both `left` / `right` and `top` / `bottom` respectively
- Added new end value scale system (`1`, `2`, `3 (default)`, `4`, `5`) that's intended to replace the `half` / `double` end values (but they are not replaced yet)

## Screenshots
Screen Shot 2020-06-10 at 12 03 30 PM

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [x] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ This is an example of a brief overview of the _Major_ or _Minor_ version changes
 - TextInput: 
   - `hasError` prop added to enable the input error state without passing `errorText`
   - `isLabelBold` prop added to allow label to be bolded
+- Added `x` and `y` directions to spacing classes (e.g. `margin--y`)
+- Added new end value for spacing classes: `1`, `2`, `3 (default)`, `4`, and `5`, which correspond to pixel values
```
Comment (akdetrick) [inline CHANGELOG.md:39]:
```suggestion
- Added new end value for spacing classes: `xs`, `s`, `m (default)`, `l`, and `xl`, which correspond to pixel values
```

File: src/base-styles/helperClasses/spacing/spacing.css [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+/*
+ - type:      margin | spacing
+ - direction: left | right | bottom | top | all | x | y
+ - scale:     1 | 2 | 3 | 4 | 5
+*/
+
+.margin--left--xs,
+.margin--x--xs,
+.margin--all--xs { margin-left: var(--space-xs); }
```
Comment (akdetrick) [inline src/base-styles/helperClasses/spacing/spacing.css:14]: this is a nice way to structure these rules 🎉

File: src/base-styles/helperClasses/spacing/spacing.css [hunk 3]
```diff
@@ -0,0 +1,178 @@
+/* Spacing classes legend
+   ======================
+   .type--direction--value
+   ====================== */
+
+/*
+ - type:      margin | spacing
+ - direction: left | right | bottom | top | all | x | y
+ - scale:     1 | 2 | 3 | 4 | 5
```
Comment (akdetrick) [inline src/base-styles/helperClasses/spacing/spacing.css:9]:
```suggestion
 - scale:     xs | s | m | l | xl
```

Comment (reywright) [reply src/base-styles/helperClasses/spacing/spacing.css]: good catch!

File: CHANGELOG.md (hunk 1 above)
Comment (reywright) [reply CHANGELOG.md]: thanks for catch this

**Pull request merged**